I spot some test fails on u85 that needs to behnadled before merge to not break CI
```
FAILED backends/arm/test/ops/test_mean_dim.py::test_mean_dim_a16w8_u85_INT[rank_1_keepdim] - AssertionError: Output 0 does not match reference output.
	Given atol: 0.002953125, rtol: 0.001.
	Output tensor shape: torch.Size([1]), dtype: torch.float32
	Difference: max: -0.416595458984375, abs: 0.416595458984375, mean abs error: 0.416595458984375.
	-- Model vs. Reference --
	 Numel: 1, 1
	Median: 0.0, 0.416595458984375
	  Mean: 0.0, 0.416595458984375
	   Max: 0.0, 0.416595458984375
	   Min: 0.0, 0.416595458984375
FAILED backends/arm/test/ops/test_mean_dim.py::test_mean_dim_a16w8_u85_INT[rand_1_keepdim] - AssertionError: Output 0 does not match reference output.
	Given atol: 0.003650776645168662, rtol: 0.001.
	Output tensor shape: torch.Size([1, 1, 7, 3]), dtype: torch.float32
	Difference: max: -0.2187926173210144, abs: 0.6785573959350586, mean abs error: 0.48883259509290966.
	-- Model vs. Reference --
	 Numel: 21, 21
	Median: 0.0, 0.5271524786949158
	  Mean: 0.0, 0.48883259509290966
	   Max: 0.0, 0.6785573959350586
	   Min: 0.0, 0.2187926173210144
FAILED backends/arm/test/ops/test_mean_dim.py::test_mean_dim_a16w8_u85_INT[rank_1] - AssertionError: Output 0 does not match reference output.
	Given atol: 0.002953125, rtol: 0.001.
	Output tensor shape: torch.Size([]), dtype: torch.float32
	Difference: max: -0.416595458984375, abs: 0.416595458984375, mean abs error: 0.416595458984375.
	-- Model vs. Reference --
	 Numel: 1, 1
	Median: 0.0, 0.416595458984375
	  Mean: 0.0, 0.416595458984375
	   Max: 0.0, 0.416595458984375
	   Min: 0.0, 0.416595458984375
FAILED backends/arm/test/ops/test_mean_dim.py::test_mean_dim_a16w8_u85_INT[rand_3] - AssertionError: Output 0 does not match reference output.
	Given atol: 0.004313722088932991, rtol: 0.001.
	Output tensor shape: torch.Size([1, 5, 7]), dtype: torch.float32
	Difference: max: -0.15074846148490906, abs: 0.8482610583305359, mean abs error: 0.48883466039385115.
	-- Model vs. Reference --
	 Numel: 35, 35
	Median: 0.0, 0.50409996509552
	  Mean: 0.0, 0.48883466039385115
	   Max: 0.0, 0.8482610583305359
	   Min: 0.0, 0.15074846148490906
```

--- a/backends/arm/test/ops/test_mean_dim.py
+++ b/backends/arm/test/ops/test_mean_dim.py
@@ -6,6 +6,7 @@
 # LICENSE file in the root directory of this source tree.
 from typing import Callable
 
+import pytest
 import torch
 from executorch.backends.arm.test import common
 from executorch.backends.arm.test.tester.test_pipeline import (
@@ -391,5 +392,70 @@ def test_mean_tosa_INT(test_data):
         test_data(),
         [],  # Might be sum, avgpool, or both
         symmetric_io_quantization=True,
+    )
+    pipeline.run()
+
+
+a16w8_mean_test_parameters = {
+    "rank_2_keepdim": lambda: (torch.rand(7, 3), (0, 1), True),
+    "rank_3_keepdim": lambda: (torch.rand(5, 7, 3), (0, 1, 2), True),
+    "rand_23_keepdim": lambda: (torch.rand(1, 5, 7, 3), (2, 3), True),
+    "rand_0123_keepdim": lambda: (torch.rand(1, 5, 7, 3), (0, 1, 2, 3), True),
+    "rand_none_keepdim": lambda: (torch.rand(1, 5, 7, 3), None, True),
+    "rank_2": lambda: (torch.rand(5, 7), (-2, -1), False),
+    "rand_123": lambda: (torch.rand(1, 5, 7, 3), (-3, -2, -1), False),
+}
+
+a16w8_mean_test_parameters_u85_xfails = {
+    "rank_1_keepdim": lambda: (torch.rand(7), 0, True),
+    "rand_1_keepdim": lambda: (torch.rand(1, 5, 7, 3), (1), True),
+    "rank_1": lambda: (torch.rand(7), (-1), False),
+    "rand_3": lambda: (torch.rand(1, 5, 7, 3), (-1), False),
+}
+
+
+@common.parametrize(
+    "test_data", {**a16w8_mean_test_parameters, **a16w8_mean_test_parameters_u85_xfails}
+)
+@common.XfailIfNoCorstone300
+def test_mean_dim_a16w8_u55_INT(test_data):
+    test_data, dim, keep_dim = test_data()
+    pipeline = EthosU55PipelineINT[input_t](
+        MeanDim(dim, keep_dim),
+        (test_data,),
+        [],
+        symmetric_io_quantization=True,
+        a16w8_quantization=True,
+    )
+    pipeline.run()
+
+
+@common.parametrize("test_data", a16w8_mean_test_parameters)
+@common.XfailIfNoCorstone320
+def test_mean_dim_a16w8_u85_INT(test_data):
+    test_data, dim, keep_dim = test_data()
+    pipeline = EthosU85PipelineINT[input_t](
+        MeanDim(dim, keep_dim),
+        (test_data,),
+        [],
+        symmetric_io_quantization=True,
+        a16w8_quantization=True,
+    )
+    pipeline.run()
+
+
+@common.parametrize("test_data", a16w8_mean_test_parameters_u85_xfails)
+@pytest.mark.xfail(
+    reason="U85 mean_dim a16w8 produces all-zero output for single-dim reductions"
+)
+@common.XfailIfNoCorstone320
+def test_mean_dim_a16w8_u85_INT_xfail(test_data):
+    test_data, dim, keep_dim = test_data()
+    pipeline = EthosU85PipelineINT[input_t](
+        MeanDim(dim, keep_dim),
+        (test_data,),
+        [],
+        symmetric_io_quantization=True,
+        a16w8_quantization=True,
     )
     pipeline.run()

--- a/backends/arm/test/targets.bzl
+++ b/backends/arm/test/targets.bzl
@@ -38,6 +38,7 @@ def define_arm_tests():
         "ops/test_to_copy.py",
         "ops/test_exp.py",
         "ops/test_reciprocal.py",
+        "ops/test_mean_dim.py",
     ]
 
     # Quantization

